Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp

```diff
@@ -4131,3 +4131,81 @@ WebKitFeatureList* webkit_settings_get_development_features(void)
 {
     return webkitFeatureListCreate(WebPreferences::internalDebugFeatures());
 }
+
+/**
+ * webkit_settings_apply_from_key_file:
+ * @settings: a #WebKitSettings
+ * @key_file: a #GKeyFile
+ * @group_name: Name of the group to read from @key_file
```

Comment:
Why limiting to a group? If we have a key file with several groups we need to call this for every group?

Reply:
Yes. For instance you could have 2 groups in the same file, one for prod, one for debugging and depending on user's choice, either one of the groups would be loaded.

Reply:
The group may contain any setting, the idea is that you can have an INI-like file to configure your browser, and populate the settings from one of its sections. Like this:

```ini
[options]
homepage = http://webkitgtk.org
download-path = ~/Downloads

[webkit-settings]
enable-developer-extras = true
enable-resizable-text-areas = true
```

By allowing indicating the `web-settings` group we allow settings loaded by `webkit_settings_apply_from_key_file()` to coexist with other program settings, without forcing a particular group name.

For example in the future we could add `webkit_settings_apply_features_from_key_file()` and the same INI file could have a `[webkit-features]` group, as well.

Reply:
I see, both use cases make sense indeed.